### ISSUE TITLE
Abstract `download_file` protocols across devices due to similarity

### ISSUE THREAD

**Issue by davidverweij** [ideafast/middleware-services#24]
Just a note for now, but this is very similar to the dreem implementation (which is fine), so perhaps in a future PR we can abstract it out into a shared task rather than device specific? since the only difference between dreem/VTT at the moment is how data is downloaded (line 103).

This is food for thought on how we might abstract this further once we implement the DAGs

_Originally posted by @jawrainey in https://github.com/ideafast/middleware-services/pull/21#discussion_r568535114_

**Comment by davidverweij** [comment]
This now also includes other protocols, such as the recently introduced `uid_to_hash()` in combination with `all_hashes()` methods.

**Comment by jawrainey** [comment]
Agreed -- also pulling out device agnostic logic such as `__device_id_from_ucam` in `dreem.py` since it determines a device in UCAM by a wear period and DeviceType (needs renaming too).

**Comment by jawrainey** [comment]
Worth noting now that while there is some overlap, BTF differs in that it returns a filename rather than bool. **I argue that DRM and other pipelines should change to follow BTF for consistency.**
